### ISSUE TITLE
测试结果不一致

### ISSUE THREAD

**Issue by AAAAA-pn** [xuguangning1218/Typhoon_Path#8]
作者你好，首先非常感谢您对源代码的提供。有一个问题想请教关于多次运行模型的测试结果有很大的差异请问是什么原因？

第一次运行结果
2015 年:                                                                        
avg lat: 0.7126403963513318
avg long: 0.9117299033681724
avg distance error: 131.59809841659265
2016 年:
avg lat: 0.9145567076825422
avg long: 0.8991142007714638
avg distance error: 149.36807381081442
2017 年:
avg lat: 0.842439612045007
avg long: 1.0521149495068716
avg distance error: 151.66574267926066
2018 年:
avg lat: 0.8523108818394957
avg long: 1.0597147503798414
avg distance error: 154.1165352155157
  第二次运行结果：
2015 年:
avg lat: 0.7671138476695258
avg long: 1.0444474997499424
avg distance error: 144.52759043638247
2016 年:
avg lat: 0.8554056742927537
avg long: 0.9066147193830667
avg distance error: 144.20323567376613
2017 年:
avg lat: 0.960007579887615
avg long: 1.220092683679917
avg distance error: 173.8872063337569
2018 年:
avg lat: 0.9279201718181292
avg long: 1.3145606559204401
avg distance error: 177.8696864267003

**Comment by xuguangning1218** [comment]
您好，感谢对工作的关注。

请您检查是否调用model.eval()，如果希望训练结果每次一致，请设置随机种子，我当时并没有设置，例如：

def seed_everything(seed: int):
    import random, os
    import numpy as np
    import torch
    
    random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = True

**Comment by AAAAA-pn** [comment]
感谢作者的回答 我现在已经拥有era5的z的在分析数据 
我想请问作者图片中您所提供的再分析数据是通过怎样的预处理所得到的


![Image](https://github.com/user-attachments/assets/a71de9b4-3285-4060-8842-4ebc3a849385)

**Comment by xuguangning1218** [comment]
感谢关注。具体预处理方法的介绍请参考论文《AM-ConvGRU: a spatio-temporal model for typhoon path prediction》的3.4 Dataset and preprocessing method部分
